### PR TITLE
Ensure cursored LRv2 calls are dispatched to LRv2

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

**Review by tstirrat15 — APPROVED** [review]
LGTM

File: pkg/cursor/cursor.go [hunk 1]
```diff
@@ -60,34 +60,51 @@ func EncodeFromDispatchCursor(dispatchCursor *dispatch.Cursor, callAndParameterH
...
 	if v1decoded.CallAndParametersHash != callAndParameterHash {
-		return nil, NewInvalidCursorErr(ErrHashMismatch)
+		return nil, nil, NewInvalidCursorErr(ErrHashMismatch)
 	}
 
 	return &dispatch.Cursor{
 		DispatchVersion: v1decoded.DispatchVersion,
 		Sections:        v1decoded.Sections,
-	}, nil
+	}, v1decoded.Flags, nil
```
Comment (ecordell) [inline pkg/cursor/cursor.go:107]: Why not include the flags in `dispatch.Cursor`? Why return them separately?

Comment (josephschorr) [reply pkg/cursor/cursor.go:107]: Because they aren't cursor dispatch flags, they are *API* dispatch flags

**Review by vroldanbet — APPROVED** [review]
Looks good to me!

**Pull request merged**